File: src/main/scala/at/pria/koza/polybuf/PolybufException.java
/**
 * PolybufException.java
 * 
 * Created on 18.05.2013
 */

package at.pria.koza.polybuf;


import java.io.IOException;


/**
 * <p>
 * {@code PolybufException}
 * </p>
 * 
 * @version V1.0 18.05.2013
 * @author SillyFreak
 */
public class PolybufException extends IOException {
    private static final long serialVersionUID = -2257893361939230762L;
    
    public PolybufException() {}
    
    public PolybufException(String message, Throwable cause) {
        super(message, cause);
    }
    
    public PolybufException(String message) {
        super(message);
    }
    
    public PolybufException(Throwable cause) {
        super(cause);
    }
}

File: src/main/scala/at/pria/koza/polybuf/PolybufException.scala
/**
 * PolybufException.scala
 *
 * Created on 18.05.2013
 */

package at.pria.koza.polybuf

import java.io.IOException

/**
 * <p>
 * `PolybufException`
 * </p>
 *
 * @version V1.0 18.05.2013
 * @author SillyFreak
 */
class PolybufException(msg: String, cause: Throwable) extends IOException(msg, cause) {
  def this() = this(null, null)
  def this(msg: String) = this(msg, null)
  def this(cause: Throwable) = this(if (cause == null) null else cause.toString(), cause)
}
